### PR TITLE
[Enhance] Add waymo unit test

### PR DESCRIPTION
**Pull request by Wuziyi616**
Fix [issue#447](https://github.com/open-mmlab/mmdetection3d/issues/447). I add some basic unit tests for WaymoDataset. Mainly about `getitem` and `show` (because I want to check its compatibility in dataset_browse). I am not familiar with Waymo format of box and its `evaluate` function. Maybe someone can help me continue this PR?

### CODE REVIEW

**Comment by ZwwWayne** [comment]
The file sizes for unit tests are too big. We need to make the size of test data smaller. See other files for unit test.

**Comment by Wuziyi616** [comment]
> The file sizes for unit tests are too big. We need to make the size of test data smaller. See other files for unit test.

Yes I have made them smaller (each <1MB).

**Comment by Wuziyi616** [comment]
I have some trouble in the Waymo protocol testing of `test_evaluate()` in `test_waymo_dataset.py`. I got expected mAP in the KITTI protocol, but wrong results in the Waymo protocol (all 0). The GT bbox I generated in `gt.bin` is the same as the predicted bbox after kitti_to_waymo conversion ([here](https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/core/evaluation/waymo_utils/prediction_kitti_to_waymo.py#L198-L199)). I don't know why the results are wrong...

File: tests/test_data/test_datasets/test_waymo_dataset.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    if not torch.cuda.is_available():
+        pytest.skip('test requires GPU and torch+cuda')
+    from mmdet3d.core.bbox import LiDARInstance3DBoxes
+    data_root, ann_file, classes, pts_prefix, pipeline, \
+        modality, split = _generate_waymo_val_dataset_config()
+    waymo_dataset = WaymoDataset(data_root, ann_file, split, pts_prefix,
+                                 pipeline, classes, modality)
+    boxes_3d = LiDARInstance3DBoxes(
+        torch.tensor(
+            # [[31.9723, -2.3247, 2.2145, 2.3900, 6.0800, 3.1000, -1.3200]]))
```
Comment (Tai-Wang) [inline tests/test_data/test_datasets/test_waymo_dataset.py:134]: Remove unused comments

**Comment by Tai-Wang** [comment]
> I have some trouble in the Waymo protocol testing of `test_evaluate()` in `test_waymo_dataset.py`. I got expected mAP in the KITTI protocol, but wrong results in the Waymo protocol (all 0). The GT bbox I generated in `gt.bin` is the same as the predicted bbox after kitti_to_waymo conversion ([here](https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/core/evaluation/waymo_utils/prediction_kitti_to_waymo.py#L198-L199)). I don't know why the results are wrong...
> 
> The predicted bbox after conversion:
> 
> ```
> objects {
>   object {
>     box {
>       center_x: 69.67650333936074
>       center_y: 33.340902794888315
>       center_z: 0.7713991550544462
>       width: 2.009999990463257
>       length: 4.360000133514404
>       height: 1.4600000381469727
>       heading: -1.4807963232186179
>     }
>     type: TYPE_VEHICLE
>   }
>   score: 0.5
>   context_name: "1071392229495085036_1844_790_1864_790"
>   frame_timestamp_micros: 1507315488219118
> }
> ```
> 
> The loaded `gt.bin` file GT bbox:
> 
> ```
> objects {
>   object {
>     box {
>       center_x: 69.67650333936074
>       center_y: 33.340902794888315
>       center_z: 0.7713991550544462
>       width: 2.009999990463257
>       length: 4.360000133514404
>       height: 1.4600000381469727
>       heading: -1.4807963232186179
>     }
>     type: TYPE_VEHICLE
>   }
>   score: 1.0
>   context_name: "1071392229495085036_1844_790_1864_790"
>   frame_timestamp_micros: 1507315488219118
>   camera_name: FRONT
> }
> ```
Sometimes evaluation methods of these datasets may not deal with all the corner cases. You may try to validate it with more samples or add some noises to these predictions.

File: tests/test_data/test_datasets/test_waymo_dataset.py (hunk 1 above)
Comment (Wuziyi616) [reply tests/test_data/test_datasets/test_waymo_dataset.py:134]: Done.

**Comment by Wuziyi616** [comment]
I finally solved the problem and this PR is ready for review now @ZwwWayne!

**Review by Tai-Wang — APPROVED** [review]

**Review by ZwwWayne — APPROVED** [review]

**Review by ZwwWayne — APPROVED** [review]
